subnet/local_manager.go

```diff
@@ -344,3 +364,126 @@ func isSubnetConfigCompat(config *Config, sn ip.IP4Net) bool {
 
 	return sn.PrefixLen == config.SubnetLen
 }
+
+func (m *LocalManager) tryAddReservation(ctx context.Context, network string, r *Reservation) error {
+	attrs := &LeaseAttrs{
+		PublicIP: r.PublicIP,
+	}
+
+	_, err := m.registry.createSubnet(ctx, network, r.Subnet, attrs, 0)
+	switch {
+	case err == nil:
+		return nil
+
+	case !isErrEtcdNodeExist(err):
+		return err
+	}
+
+	// This subnet or its reservation already exists.
+	// Get what's there and
+	// - if PublicIP matches, remove the TTL make it a reservation
+	// - otherwise, error out
+	sub, asof, err := m.registry.getSubnet(ctx, network, r.Subnet)
+	switch {
+	case err == nil:
+	case isErrEtcdNodeExist(err):
```

Comment:
Oh yeah, I remember why I don't have a test for it -- it's very difficult to write as I would need to perfectly time a race.